deep_quoridor/src/utils/misc.py

```diff
@@ -35,3 +38,81 @@ def my_device():
         return torch.device("mps")
 
     return torch.device("cpu")
+
+
+def yargs(parser: argparse.ArgumentParser, default_path: str):
+    """
+    This function enhances the command line argument parsing by allowing arguments to be
+    specified in YAML configuration files. It supports both single file and directory inputs,
+    where multiple YAML configurations can be defined.
+
+    Args:
+        parser (argparse.ArgumentParser): The argument parser to augment with YAML support
+        default_path (str): The default path to look for YAML configuration files
+
+    Returns:
+        dict: A dictionary mapping configuration IDs to their parsed arguments. If no YAML
+            configuration is used, returns a dictionary with an empty string key and the
+            parsed command line arguments as value.
+
+    The function adds two optional arguments to the parser:
+        -yp/--yargs_path: Path to the YAML configuration file or directory (if none, the default_path will be used)
+        -yi/--yargs_ids: List of configuration IDs to run (if None, runs all configs)
+
+    Example yaml file:
+        benchmark_B5W3:
+            args: -t 100 -N 5 -W 3 --players greedy greedy:p_random=0.1,nick=greedy-ish
+        benchmark_B9W10:
+            args: -t 100 -N 9 -W 10 --players greedy greedy:p_random=0.1,nick=greedy-ish
+
+    Example calls from the command line:
+        # Execute all the entries in all the yaml files in default_path
+        python myapp.py -yp
+
+        # Execute all the entries in the file {default_path}/agent_benchmark.yaml
+        python myapp.py -yp agent_benchmark.yaml
+
+        # Execute the entries benchmark_B5W3 benchmark_B9W10 (should be found in the files in the default_path)
+        python myapp.py -yi benchmark_B5W3 benchmark_B9W10
+
+    """
+    parser.add_argument(
+        "-yp",
+        "--yargs_path",
```

Comment:
nit: posix style for long args is to use dashes instead of underscores, so this would be `--yargs-path`: https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

argparse is smart enough to convert these so that the resulting argument is `args.yargs_path` in python, so that the python variable follows python conventions and the command line arg follows posix conventions.